models/schedule_item.go

```diff
@@ -66,11 +68,101 @@ func NewTimeslotItem(t *myradio.Timeslot, finish time.Time, u func(*myradio.Time
+	} else if s.Contains(name, s.ToLower("ury lunch")) {
+		return "flagship"
+	} else if s.Contains(name, s.ToLower("ury brunch")) {
+		return "flagship"
+	} else if s.Contains(name, s.ToLower("URY Brunch")) {
+		return "flagship"
+	} else if s.Contains(name, s.ToLower("URY Afternoon Tea:")) {
+		return "flagship"
+	} else if s.Contains(name, s.ToLower("URY:PM")) {
+		return "flagship"
+	} else if s.Contains(name, s.ToLower("ury news")) {
+		return "news"
+	} else if s.Contains(name, s.ToLower("ury sports")) {
+		return "news"
+	} else if s.Contains(name, s.ToLower("ury football")) {
+		return "news"
+	} else if s.Contains(name, s.ToLower("york sport report")) {
+		return "news"
+	} else if s.Contains(name, s.ToLower("university radio talk")) {
+		return "news"
+	} else if s.Contains(name, s.ToLower("candidate interview night")) {
+		return "news"
+	} else if s.Contains(name, s.ToLower("election results night")) {
+		return "news"
+	} else if s.Contains(name, s.ToLower("yusu election")) {
+		return "news"
+	} else if s.Contains(name, s.ToLower("The Second Half With Josh Kerr")) {
+		return "news"
+	} else if s.Contains(name, s.ToLower("URY SPORT")) {
+		return "news"
+	} else if s.Contains(name, s.ToLower("ury speech")) {
+		return "speech"
+	} else if s.Contains(name, s.ToLower("yorworld")) {
+		return "speech"
+	} else if s.Contains(name, s.ToLower("in the stalls")) {
+		return "speech"
+	} else if s.Contains(name, s.ToLower("screen")) {
+		return "speech"
+	} else if s.Contains(name, s.ToLower("stage")) {
+		return "speech"
+	} else if s.Contains(name, s.ToLower("game breaking")) {
+		return "speech"
+	} else if s.Contains(name, s.ToLower("radio drama")) {
+		return "speech"
+	} else if s.Contains(name, s.ToLower("Book Corner")) {
+		return "speech"
+	} else if s.Contains(name, s.ToLower("Saturated Facts")) {
+		return "speech"
+	} else if s.Contains(name, s.ToLower("URWatch")) {
+		return "speech"
+	} else if s.Contains(name, s.ToLower("Society Challenge")) {
+		return "speech"
+	} else if s.Contains(name, s.ToLower("Speech Showcase")) {
+		return "speech"
+	} else if s.Contains(name, s.ToLower("roses live 201")) {
+		return "event"
+	} else if s.Contains(name, s.ToLower("roses 201")) {
+		return "event"
+	} else if s.Contains(name, s.ToLower("woodstock")) {
+		return "event"
+	} else if s.Contains(name, s.ToLower("movember")) {
+		return "event"
+	} else if s.Contains(name, s.ToLower("panto")) {
+		return "event"
+	} else if s.HasPrefix(name, s.ToLower("101:")) {
+		return "event"
+	} else if s.Contains(name, s.ToLower("Vanbrugh Chair Debate")) {
+		return "event"
+	} else if s.Contains(name, s.ToLower("URY Does RAG Courtyard Takeover")) {
+		return "event"
+	} else if s.Contains(name, s.ToLower("URY Presents")) {
+		return "event"
+	} else if s.Contains(name, s.ToLower("URYOnTour")) {
+		return "event"
+	} else if s.Contains(name, s.ToLower("URY On Tour")) {
+		return "event"
+	} else if (StartTime.Hour() >= 21) || (StartTime.Hour() < 5) {
```

Comment:
It's difficult to see where one type of block ends and another type of block begins.  When I wrote the comment above I didn't realise that this was also doing range-based block matching.

Again, I'm not convinced this is 2016-site's battle.  MyRadio should be doing this IMO, but the unsolved question is how MyRadio should be calculating it, and how MyRadio should be exposing it.